mlx/primitives.cpp

```diff
@@ -1338,6 +1338,88 @@ std::pair<array, int> LogicalNot::vmap(
   return {logical_not(inputs[0], stream()), axes[0]};
 }
 
+std::vector<array> LogicalAnd::vjp(
+    const std::vector<array>& primals,
+    const array& cotan,
+    const std::vector<int>& argnums) {
+  assert(primals.size() == 2);
+
+  if (argnums.size() == 1) {
+    // If derivative is with respect to only one argument
+    return {argnums[0] == 0 ? cotan : zeros_like(cotan, stream())};
```

Comment:
I don't think it's correct to return `cotan` here. You should return `zeros` in every case.